src/components/CodeView/index.spec.tsx

```diff
@@ -65,4 +100,72 @@ describe(__filename, () => {
     );
     expect(root.find(`.${styles.code}`)).toHaveLength(contentLines.length);
   });
+
+  it('renders an HTML ID for each line', () => {
+    const root = render({ content: 'line 1\nline 2' });
+
+    expect(root.find(`.${styles.line}`)).toHaveLength(2);
+    expect(root.find(`.${styles.line}`).at(0)).toHaveProp('id', 'L1');
+    expect(root.find(`.${styles.line}`).at(1)).toHaveProp('id', 'L2');
+  });
+
+  it('marks a row as selected', () => {
+    const selectedLine = 2;
+    const location = createFakeLocation({ hash: `#L${selectedLine}` });
+
+    const root = render({ content: 'line 1\nline 2', location });
+
+    expect(root.find(`.${styles.selectedLine}`)).toHaveLength(1);
+  });
+
+  it('renders a link for each line number', () => {
+    const root = render({ content: 'single line' });
+
+    expect(root.find(`.${styles.lineNumber}`)).toHaveLength(1);
+    expect(root.find(`.${styles.lineNumber}`).find(Link)).toHaveLength(1);
+    expect(root.find(`.${styles.lineNumber}`).find(Link)).toHaveProp(
+      'to',
+      '#L1',
+    );
+    // This is an anchor on the table row. This is a bit confusing here because
+    // `#` refers to the ID (CSS) selector and not the hash. The ID value is
+    // `L1`.
+    expect(root.find('#L1')).toHaveLength(1);
+  });
+
+  it('calls _scrollToSelectedLine() when rendering a selected line', () => {
+    const selectedLine = 2;
+    const lines = ['first', 'second'];
+    const content = lines.join('\n');
+    const location = createFakeLocation({ hash: `#L${selectedLine}` });
+    const _scrollToSelectedLine = jest.fn();
+
+    // We need `mount()` because `ref` is only used in a DOM environment.
+    const root = renderWithMount({ _scrollToSelectedLine, content, location });
+
+    expect(_scrollToSelectedLine).toHaveBeenCalledWith(
```

Comment:
I know that this relies on an execution of `ref` but I don't think that's necessary. This `ref` execution behavior is part of the React interface and won't be breaking anytime soon. You could do a shallow render and simply ensure that the `ref` prop with the expected value is passed into the right element. The drawback of using `mount()` is that all leaf components are rendered; this could make for a brittle test.

Reply:
ref does not exist outside mount, it's a special prop and enzyme does not see it when using shallow :(

Reply:
Ah, bummer. It's weird that enzyme hides the ref configuration.